devito/types/petsc.py

```diff
@@ -66,6 +66,15 @@ class PETScArray(ArrayBasic):
 
     _data_alignment = False
 
+    __rkwargs__ = (ArrayBasic.__rkwargs__ +
+                   ('liveness',))
+
+    def __init_finalize__(self, *args, **kwargs):
+        self._liveness = kwargs.pop('liveness', 'lazy')
```

Comment:
I'd replace it with:

```
kwargs.setdefault('liveness', 'lazy')
```

then drop the assert and just let `super` do its job